tests/test_integration_partition_evolution.py

```diff
@@ -0,0 +1,423 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint:disable=redefined-outer-name
+
+import pytest
+
+from pyiceberg.catalog import Catalog, load_catalog
```

Comment:
Recently, we have moved integration tests into `tests/integration`: #207 . Shall we move this file into that folder too? We can name it as `test_rest_partition_evolution.py` to indicate that these tests use RestCatalog. 

Since we've implemented `_commit_table` for Hive #294 , we could refactor `test_rest_*` to use both rest and hive catalog. Using catalog can help test pyiceberg's logic of `_commit_table`. But this can happen in a follow-up PR

Reply:
Sounds good. Since we are technically writing new metadata after this change, if you don't mind could I actually write the tests to exercise the non-rest case as well? The PR may get slightly bigger, so up to you if it's harder for review. I just want to make sure that when the change goes in, we're confident we're updating the metadata correctly in all cases.